routes/solutionsController.js

```diff
@@ -527,6 +527,95 @@ router.put("/solutionAccepted", async (req, res) => {
   }
 });
 
+//  API for updateSolutionStatus
+router.put("/updateSolutionStatus", async (req, res) => {
+  console.log("updateSolutionStatus", req.body);
+  const solution = req.body;
+  var valid_user = await new Promise((resolve, reject) => {
+    UserModel.findOne({ _id: solution.user_id }, (err, user) => {
+      if (!err) {
+        console.log("USER", user);
+        resolve(user);
+      } else {
+        reject(err);
+      }
+    });
+  });
+  if (valid_user) {
+    const updated_solution = await new Promise((resolve, reject) => {
+      SolutionModel.findOneAndUpdate(
+        { _id: solution.solution_id },
+        {
+          $set: {
+            status: `${solution.status}`,
+            accepted_by: ObjectID(solution.accepted_by),
+          },
+        },
+        { new: true }
+      )
+        .populate("user")
+        .exec((err, data) => {
+          if (!err) {
+            resolve(data);
+          } else {
+            reject(err);
+          }
+        });
+    });
+    if (updated_solution) {
```

Comment:
what is this if condition doing?

Reply:
I just use your previous update code and update the status by getting it from body